File: languages/java/src/main/java/com/bloock/sdk/entity/identity/StringAttributeDescriptor.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents a descriptor for an attribute with a string value.
 */
public class StringAttributeDescriptor extends AttributeDescriptor {

  /**
   * Constructs an StringAttributeDescriptor object with the specified parameters.
   * 
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   */
  public StringAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static StringAttributeDescriptor fromProto(
      BloockIdentityEntities.StringAttributeDefinition res) {
    return new StringAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public BloockIdentityEntities.StringAttributeDefinition toProto() {
    return BloockIdentityEntities.StringAttributeDefinition.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}

File: languages/js/src/entity/identity_v2/string_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

export class StringAttributeDescriptor extends AttributeDescriptor {

  constructor(
    displayName: string,
    technicalName: string,
    description: string,
    required: boolean
  ) {
    super(displayName, technicalName, description, required);
  }

  public toProto(): identityEntitiesProto.StringAttributeDefinitionV2 {
    return identityEntitiesProto.StringAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.StringAttributeDefinitionV2
  ): StringAttributeDescriptor {
    return new StringAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}
